Document/0x07a-Testing-Authentication-and-Session-Management.md

```diff
@@ -66,6 +66,98 @@ If any of these two conditions raise an issue, reject the request and do not all
 
 
 
+### Testing JSON Web Token (JWT)
+
+#### Overview
+
+JSON Web Token (JWT) ensures the integrity and secure transmission of information within a JSON object between two parties and is defined in RFC 7519<sup>[1]</sup>. For mobile apps it's more and more used to authenticate both, the message sender and receiver.
+
+An example of an encoded JSON Web Token can be found below<sup>[5]</sup>.
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ
+```
+
+JWTs are Base-64 encoded and are divided into three parts:
+
+* **Header** Algorithm and Token Type (eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9):
+```JSON
+{"alg":"HS256","typ":"JWT"}
+```
+* **Claims** Data  (eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9):
+```JSON
+{"sub":"1234567890","name":"John Doe","admin":true}
+```
+* **JSON Web Signature (JWS)** (TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ):
+```JSON
+HMACSHA256(
+  base64UrlEncode(header) + "." +
+  base64UrlEncode(payload),
+  secret
+)
+```
+
+JWT implementations are available for all major programming languages, like PHP<sup>[2]</sup> or Java Spring<sup>[3]</sup>.
+
+#### Static Analysis
+
+Identify the JWT library that is used on server and client side. It should be checked if there are any known vulnerabilities available for the JWT libraries in use.
```

Comment:
> It should be checked

I think passive voice like this should be avoided. Maybe it's better written as an instruction: "Check if there are any known vulnerabilities available for the JWT libraries in use."